src/BridgedRoom.ts

```diff
@@ -0,0 +1,886 @@
+                name: this.slackChannelName,
+                slack_bot_id: this.slackBotId,
+                slack_bot_token: this.slackBotToken,
+                slack_team_domain: this.slackTeamDomain,
+                slack_team_id: this.slackTeamId,
+                slack_user_id: this.slackUserId,
+                slack_user_token: this.slackUserToken,
+                webhook_uri: this.slackWebhookUri,
+            },
+            remote_id: this.inboundId,
+        };
+        this.dirty = false;
+        return entry;
+    }
+
+    public async onMatrixReaction(message: any) {
+        if (!this.SlackBotToken) { return; }
+
+        const relatesTo = message.content["m.relates_to"];
+        const eventStore = this.main.eventStore;
+        const event = await eventStore.getEntryByMatrixId(message.room_id, relatesTo.event_id);
+
+        // If we don't get an event then exit
+        if (event === null) {
+            log.debug("Could not find event to react to.");
+            return;
+        }
+
+        // Convert the unicode emoji into a slack emote name
+        let emojiKeyName: string;
+        const emojiItem = emoji.find(relatesTo.key);
+        if (emojiItem !== undefined) {
+            emojiKeyName = emojiItem.key;
+        } else {
+            emojiKeyName = relatesTo.key;
+            // Strip the colons
+            if (emojiKeyName.startsWith(":") && emojiKeyName.endsWith(":")) {
+                emojiKeyName = emojiKeyName.substring(1, emojiKeyName.length - 1);
+            }
+        }
+
+        // TODO: This only works once from matrix as we are sending the event as the
+        // bot user.
+        const body = {
+            as_user: false,
+            channel: this.slackChannelId,
+            name: emojiKeyName,
+            timestamp: event.remoteEventId,
+        };
+
+        const sendMessageParams = {
+            body,
+            headers: {},
+            json: true,
+            method: "POST",
+            uri: "https://slack.com/api/reactions.add",
+        };
+
+        if (this.slackBotToken) {
+            sendMessageParams.headers = {
+                Authorization: "Bearer " + this.slackBotToken,
+            };
+        }
+
+        const res = await rp(sendMessageParams);
+        if (!res || !res.ok) {
+            log.error("HTTP Error: ", res);
+        } else {
+            // TODO: Add this event to the event store
+            // Unfortunately reactions.add does not return the ts of the reactions event.
+            // So we can't store it in the event store
+        }
+        return res;
+
+    }
+
+    public async onMatrixRedaction(message: any) {
+        if (!this.slackBotToken) { return; }
+        const event = await this.main.eventStore.getEntryByMatrixId(message.room_id, message.redacts);
+
+        // If we don't get an event then exit
+        if (event === null) {
+            log.debug("Could not find event '${message.redacts}' in room '${message.room_id}' to delete.");
```

Comment:
Is the fact this isn't a backtick type string deliberate or just not a fix that got pulled over from develop?

Reply:
I forgot to fix this :/